ingest/ingest_pipeline.py

```diff
@@ -191,22 +191,29 @@ def parse_arguments():
     parser_ingest_expression.add_argument('--gene-file',
                                           help='Path to .genes.tsv file')
 
-    parser_ingest_cluster = subargs.add_parser('ingest_clusters',
+    parser_ingest_cluster = subparser.add_parser('ingest_clusters',
                                                help='Indicates that cluster'
                                                ' files are being ingested')
     parser_ingest_cluster.add_argument('--cluster-file',
                                        help='Path to cluster files')
-    parsed_args = args.parse_args()
+    return parser
+
+def validate_arguments(parsed_args):
```

Comment:
Here you're mock the args argument. Then you'd set the return values of parsed_args.matrix_file_type.return_value to an incorrect value.  Then you'd check to see if the assertion error was raised with assertRaises.

Reply:
You'd also test for the positive test case here as well.

Reply:
Sounds good, I aim to do this today after demo.